server/api/db/sqldb/db.py

```diff
@@ -840,18 +840,26 @@ def list_artifact_tags(
         """
         :return: a list of Tuple of (project, artifact.key, tag)
         """
-        artifacts = self.list_artifacts(session, project=project, category=category)
+
+        # query artifact key and tag from the db
+        query = (
+            self._query(session, ArtifactV2, project=project)
+            .join(ArtifactV2.Tag, ArtifactV2.Tag.obj_id == ArtifactV2.id)
+            .with_entities(ArtifactV2.key, ArtifactV2.Tag.name)
+        )
```

Comment:
will that ensure distinct tag names?
also, why not query over tags and join over artifacts? because then the join will start with less entities to cross over

Reply:
updated the query